tf2/include/tf2/time_cache.h

```diff
@@ -117,13 +126,14 @@ class TimeCache : public TimeCacheInterface
   TF2_PUBLIC
   virtual bool getData(
     tf2::TimePoint time, tf2::TransformStorage & data_out,
-    std::string * error_str = 0);
+    TimeCacheError * error_code = 0, std::string * error_str = 0);
```

Comment:
I perfer to include this new argument at the end, this way we are not breaking all users. @clalancette thoughts ?

Reply:
Yes, agreed; we should keep backwards compatibility where we can, and this seems like an easy one to do it on.

Reply:
No problem! I can do this :) 
Tomorrow probably (now I'm on mobile)

Reply:
See next commit eea423c